File: python/pages/alerts_page.py
from selenium.webdriver.common.by import By
from .base_page import BasePage


class AlertsPage(BasePage):

    ALERT_BTN = (By.ID, "alert-btn")
    CONFIRM_BTN = (By.ID, "confirm-btn")
    PROMPT_BTN = (By.ID, "prompt-btn")

    def click_alert(self):
        self.click(self.ALERT_BTN)
        self.accept_alert()
        return self

    def click_confirm_accept(self):
        self.click(self.CONFIRM_BTN)
        self.accept_alert()
        return self

    def click_prompt(self, text):
        self.click(self.PROMPT_BTN)
        alert = self.driver.switch_to.alert
        alert.send_keys(text)
        alert.accept()
        return self

File: src/test/java/com/example/pages/AlertsPage.java
package com.example.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class AlertsPage extends SeleniumBasePage {

    private final By alertButton = By.id("alert-btn");
    private final By confirmButton = By.id("confirm-btn");
    private final By promptButton = By.id("prompt-btn");

    public AlertsPage(WebDriver driver) {
        super(driver);
    }

    public AlertsPage clickAlert() {
        click(alertButton);
        acceptAlert();
        return this;
    }

    public AlertsPage clickConfirmAccept() {
        click(confirmButton);
        acceptAlert();
        return this;
    }

    public AlertsPage clickPrompt(String input) {
        click(promptButton);
        driver.switchTo().alert().sendKeys(input);
        acceptAlert();
        return this;
    }
}
